compiler/next/include/chpl/uast/Serial.h

```diff
@@ -48,35 +49,37 @@ namespace uast {
  */
 class Serial final : public Expression {
  private:
-  Serial(ASTList stmts, int8_t condChildNum, bool usesImplicitBlock)
-    : Expression(asttags::Serial, std::move(stmts)),
+  Serial(ASTList children, int8_t condChildNum, BlockStyle blockStyle)
+    : Expression(asttags::Serial, std::move(children)),
       condChildNum_(condChildNum),
-      usesImplicitBlock_(usesImplicitBlock) {
+      blockStyle_(blockStyle) {
     assert(isExpressionASTList(children_));
   }
 
   bool contentsMatchInner(const ASTNode* other) const override;
   void markUniqueStringsInner(Context* context) const override;
   int8_t condChildNum_;
```

Comment:
Is there a particular reason for not wanting to use `condition()` or is it just stylistic preference? It seems to me like `condition()` being `nullptr` says the same thing.

Reply:
FWIW I like the nullptr way. Otherwise I think we are implying that `condition` cannot return nullptr and calling it with `hasCondition() == false` is an error.

Reply:
The docs for `condition()` already say they will return `nullptr` if the condition does not exist. I am personally not a fan of using `hasThingy()` when `thingy() == nullptr` or `!thingy()` will do. Since this is a style thing, I could put a TODO above the method so that we can return to it later (similar to the `BlockStyle` enum).

Reply:
Another justification for this is that we often like to write code of the form:

```cpp
if (const Expression* cond = serial->condition()) {

}
```

And I can't imagine we'll want to stop doing that in favor of:

```cpp
if (serial->hasCondition()) {
  const Expression* cond = serial->condition();
}
```